src/front-door/azext_front_door/custom.py

```diff
@@ -559,23 +559,47 @@ def create_fd_routing_rules(cmd, resource_group_name, front_door_name, item_name
-        c.update_param('forwarding_protocol', forwarding_protocol, False)
-        c.update_param('backend_pool', SubResource(id=backend_pool) if backend_pool else None, False)
-        c.update_param('enabled_state', enabled, False)
-    with UpdateContext(instance.cache_configuration) as c:
-        c.update_param('dynamic_compression', dynamic_compression, False)
-        c.update_param('query_parameter_strip_directive', query_parameter_strip_directive, False)
-    return instance
+    if instance:
+        if hasattr(instance.route_configuration, 'forwarding_protocol'):
+            with UpdateContext(instance) as c:
+                c.update_param('frontend_endpoints', [SubResource(id=x) for x in frontend_endpoints]
+                               if frontend_endpoints else None, False)
+                c.update_param('accepted_protocols', accepted_protocols, False)
+                c.update_param('patterns_to_match', patterns_to_match, False)
+                c.update_param('enabled_state', enabled, False)
+            with UpdateContext(instance.route_configuration) as c:
+                c.update_param('custom_forwarding_path', custom_forwarding_path, False)
+                c.update_param('forwarding_protocol', forwarding_protocol, False)
+                c.update_param('backend_pool', SubResource(id=backend_pool) if backend_pool else None, False)
+            if caching:
+                with UpdateContext(instance.route_configuration.cache_configuration) as c:
+                    c.update_param('dynamic_compression', dynamic_compression, False)
+                    c.update_param('query_parameter_strip_directive', query_parameter_strip_directive, False)
+            else:
+                if hasattr(instance.route_configuration, 'cache_configuration'):
+                    instance.route_configuration.cache_configuration = None
+        elif hasattr(instance.route_configuration, 'redirect_protocol'):
+            with UpdateContext(instance) as c:
+                c.update_param('frontend_endpoints', [SubResource(id=x) for x in frontend_endpoints]
+                               if frontend_endpoints else None, False)
+                c.update_param('accepted_protocols', accepted_protocols, False)
+                c.update_param('patterns_to_match', patterns_to_match, False)
+                c.update_param('enabled_state', enabled, False)
+            with UpdateContext(instance.route_configuration) as c:
+                c.update_param('redirect_type', redirect_type, False)
+                c.update_param('redirect_protocol', redirect_protocol, False)
+                c.update_param('custom_host', custom_host, False)
+                c.update_param('custom_path', custom_path, False)
+                c.update_param('custom_fragment', custom_fragment, False)
+                c.update_param('custom_query_string', custom_query_string, False)
+    return parent
```

Comment:
why all False?

Reply:
None of these arguments supports `''` to clear the values. It doesn't make sense to have None value for each of them.

Reply:
Will those properties updatable? I couldn't find the corresponding variables to set this property?

Reply:
Mistaken the meaning of False.